Assignment3-Java/src/main/java/ca/ciccc/BankAccount.java

```diff
@@ -0,0 +1,173 @@
+            balance -= amount;
+            transactionText += "withdrawal of $" + amount + "\n";
+        }
+        transactionCount++;
+    }
+
+    /**
+     * The transactionFee method accepts a fee amount ({@code double}) as a parameter, and applies that fee
+     * to the user's past transactions. The fee is applied once for the first transaction, twice for
+     * the second transaction, three times for the third, and so on. These fees are subtracted out
+     * from the user's overall balance. If the user's balance is large enough to afford all of the
+     * fees with greater than $0.00 remaining, the method will return {@code true}. If the balance cannot
+     * afford all of the fees or has no money left, the balance is left as 0.0 and method returns
+     * {@code false}.
+     *
+     * For example, given the following ca.ciccc.BankAccount object.
+     *
+     * ca.ciccc.BankAccount savings = new ca.ciccc.BankAccount("Jimmy");
+     * savings.deposit(10.00);
+     * savings.deposit(50.00);
+     * savings.deposit(10.00);
+     * savings.deposit(70.00);
+     *
+     * savings.transactionFee(5.00);
+     *
+     * The account would be deducted $5 + $10 + $15 + $20 for the four transactions, leaving a final
+     * balance of $90.00. The method would return true.
+     *
+     * savings.transactionFee(10.00);
+     *
+     * Then the account would be deducted $10 + $20 + $30 + $40 for the four transactions, leaving
+     * a final balance of $0.00. The method would return false.
+     *
+     * @param fee
+     * @return true if there's enough balance, otherwise false
+     */
+    public boolean transactionFee(double fee) {
+        // TODO 2: Your code goes here.
+        double totFee = 0.0;
+
+        for (int i = 1; i <= transactionCount; i++)
+        {
+            totFee += (fee * i);
+        }
+        if (balance > totFee) return true;
+
+        return false;
+    }
+
+    /**
+     * The transfer method moves money from `this` bank account to another account. The method
+     * accepts two parameters: the amount of money to transfer and the other ca.ciccc.BankAccount to accept
+     * the money.
+     * There is a $5.00 fee for transferring money, so this much must be deducted from the current
+     * account's balance before any transfer.
+     *
+     * If `this` account object does not have enough money to make full transfer, then transfer
+     * whatever money is left after the $5.00 fee is deducted. If this account has under $5.00 or amount is 0 or less,
+     * no transfer should occur and neither account's state should be modified.
+     *
+     * If any amount of money is transferred, return {@code true}. Otherwise {@code false}.
+     *
+     * @param amount
+     * @param other
+     * @return true if transferred any amount of money, otherwise false.
+     */
+    public boolean transfer(double amount, BankAccount other) {
+        // TODO 3: Your code goes here.
+        if (balance <= 0) return false;
```

Comment:
[FYI] The condition in the next line includes this condition. This line looks unnecessary.